ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -380,6 +380,26 @@ status:
     status: "False"
     type: DataObtained
 ```
+### Internal Secrets
+
+To distinguish secrets that were provided by end-users from secrets that are generated by RHTAP's components, a label `appstudio.redhat.com/internal: true` is introduced. The presence of this label indicates that the secret is intended for internal usage, and unauthorized changes to such secrets may lead to unpredictable results.
+
+This label helps to identify secrets that are meant for internal purposes, preventing accidental or unauthorized modifications that could disrupt the functioning of RHTAP's components.
```

Comment:
Can we be more explicit about how these changes will be implemented in AppStudio? Will there be changes to permissions or a controller? What are the specific actions that would be prevented?